crypto/pkcs11key/key.go

```diff
@@ -249,13 +268,173 @@ func getPublicKey(module ctx, session pkcs11.SessionHandle, privateKeyHandle pkc
 			gotExponent = true
 		}
 	}
-	if !gotModulus || !gotExponent {
-		return noKey, errors.New("public key missing either modulus or exponent")
+	if !gotModulus {
+		return noKey, errors.New("private key missing modulus")
```

Comment:
We actually look in the private key so the message might be confusing. I'm suggesting: key is missing public {modulus,exponent}.